--- a/arch/arm/boot/dts/qcom-ipq8064.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8064.dtsi
@@ -1513,7 +1513,6 @@
 			qcom,load-addr = <0x40000000>;
 			qcom,turbo-frequency;
 
-			qcom,bridge-enabled;
 			qcom,gre-enabled;
 			qcom,gre-redir-enabled;
 			qcom,gre_tunnel_enabled;
@@ -1532,12 +1531,10 @@
 			qcom,vlan-enabled;
 			qcom,wlan-dataplane-offload-enabled;
 			qcom,wlanredirect-enabled;
-			qcom,pxvlan-enabled;
 			qcom,vxlan-enabled;
 			qcom,match-enabled;
 			qcom,mirror-enabled;
-			qcom,rmnet-enabled;
-			qcom,clmap-enabled;
+			qcom,tstamp-enabled;
 		};
 
 		nss1: nss@40800000 {
