--- a/arch/arm/boot/dts/qcom-ipq8064-wg2600hp.dts
+++ b/arch/arm/boot/dts/qcom-ipq8064-wg2600hp.dts
@@ -161,9 +161,25 @@
 &gmac1 {
 	status = "okay";
 
+ 	compatible = "qcom,nss-gmac";
+ 	reg = <0x37200000 0x200000>;
+ 	interrupts = <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
+
 	phy-mode = "rgmii";
 	qcom,id = <1>;
 
+ 	qcom,pcs-chanid = <0>;
+ 	qcom,phy-mdio-addr = <4>;	/* AKRO: 4->1 */
+ 	qcom,poll-required = <0>;	/* AKRO: 0->1 */
+ 	qcom,rgmii-delay = <1>;		/* AKRO: 1->0 */
+ 	qcom,phy_mii_type = <0>;	/* AKRO: 0->8(PHY_INTERFACE_MODE_RGMII) */
+ 	qcom,emulation = <0>;
+ 	qcom,forced-speed = <1000>;
+ 	qcom,forced-duplex = <1>;
+ 	qcom,socver = <0>;
+ 	qcom,irq = <255>;
+ 	mdiobus = <&mdio0>;
+ 
 	pinctrl-0 = <&rgmii2_pins>;
 	pinctrl-names = "default";
 
@@ -179,9 +195,25 @@
 &gmac2 {
 	status = "okay";
 
+ 	compatible = "qcom,nss-gmac";
+ 	reg = <0x37400000 0x200000>;
+ 	interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
+ 
 	phy-mode = "sgmii";
 	qcom,id = <2>;
 
+ 	qcom,pcs-chanid = <1>;
+ 	qcom,phy-mdio-addr = <0>;	/* AKRO: 0->0 */
+ 	qcom,poll-required = <0>; /* no polling */
+ 	qcom,rgmii-delay = <0>;
+ 	qcom,phy_mii_type = <1>;	/* AKRO: 1->4(PHY_INTERFACE_MODE_SGMII) */
+ 	qcom,emulation = <0>;
+ 	qcom,forced-speed = <1000>;
+ 	qcom,forced-duplex = <1>;
+ 	qcom,socver = <0>;
+ 	qcom,irq = <258>;
+ 	mdiobus = <&mdio0>;
+ 
 	nvmem-cells = <&macaddr_PRODUCTDATA_0>;
 	nvmem-cell-names = "mac-address";
 
